kernel/register.cc

```diff
@@ -25,6 +25,26 @@
 #include <stdio.h>
 #include <errno.h>
 
+#ifdef YOSYS_ENABLE_ZLIB
+#include <zlib.h>
+
+PRIVATE_NAMESPACE_BEGIN
+#define GZ_BUFFER_SIZE 8192
+void decompress_gzip(const std::string &filename, std::stringstream &out)
+{
+	char buffer[GZ_BUFFER_SIZE];
+	int bytes_read;
+	gzFile gzf = gzopen(filename.c_str(), "rb");
+	while(!gzeof(gzf)) {
+		bytes_read = gzread(gzf, reinterpret_cast<void *>(buffer), GZ_BUFFER_SIZE);
+		out.write(buffer, bytes_read);
```

Comment:
Would prefer to not have to copy the entire file like this, but I had a quick look and couldn't see other way other than using boost or rolling your own implementation.

Reply:
Yes, as far as I can tell this would require much more work for a memory gain likely to be fairly small in terms of total peak memory throughout runtime. Even with Boost, I think some hacking would be needed to manage destruction of both the decompresser and the underlying file correctly. We can look at this in the future again if it proves to be an issue.